presto-parser/src/main/java/com/facebook/presto/sql/tree/ExpressionTreeRewriter.java

```diff
@@ -554,22 +559,27 @@ public Expression visitFunctionCall(FunctionCall node, Context<C> context)
         // Since OrderBy contains list of SortItems, we want to process each SortItem's key, which is an expression
         private OrderBy rewriteOrderBy(OrderBy orderBy, Context<C> context)
         {
-            ImmutableList.Builder<SortItem> sortItems = ImmutableList.builder();
-            for (SortItem sortItem : orderBy.getSortItems()) {
+            List<SortItem> rewrittenSortItems = rewriteSortItems(orderBy.getSortItems(), context);
+            if (sameElements(orderBy.getSortItems(), rewrittenSortItems)) {
+                return orderBy;
+            }
+
+            return new OrderBy(rewrittenSortItems);
+        }
+
+        private List<SortItem> rewriteSortItems(List<SortItem> sortItems, Context<C> context)
+        {
+            ImmutableList.Builder<SortItem> rewrittenSortItems = ImmutableList.builder();
+            for (SortItem sortItem : sortItems) {
                 Expression sortKey = rewrite(sortItem.getSortKey(), context.get());
                 if (sortItem.getSortKey() != sortKey) {
-                    sortItems.add(new SortItem(sortKey, sortItem.getOrdering(), sortItem.getNullOrdering()));
+                    rewrittenSortItems.add(new SortItem(sortKey, sortItem.getOrdering(), sortItem.getNullOrdering()));
                 }
                 else {
-                    sortItems.add(sortItem);
+                    rewrittenSortItems.add(sortItem);
                 }
             }
-            List<SortItem> rewrittenSortItems = sortItems.build();
-            if (sameElements(orderBy.getSortItems(), rewrittenSortItems)) {
-                return orderBy;
-            }
-
-            return new OrderBy(rewrittenSortItems);
+            return rewrittenSortItems.build();
```

Comment:
Maybe `rewriteSortItems` should be part of `rewriteOrderBy` method as it was before.  Also this code change should be part of: ` Add syntax support for aggregation_function(... ORDER BY ..) ` commit